### ISSUE TITLE
STAS Review and adjust the search results detail page for accessibility

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/MyopicVicar#966]

**Comment by AlexAvlonitis** [comment]
hi @AlOneill is this something i can help with?

**Comment by AlOneill** [comment]
Is this the correct file?
https://github.com/FreeUKGen/MyopicVicar/blob/master/app/views/search_records/show.html.erb

There are 3 tables on the page: the first looks like the one I was expecting; the other two are more puzzling. The second looks like it relates to something not yet implemented -- annotations; what is the third all about?

I ask because the markup is not semantic -- do I need to make all 3 tables semantic/accessible, or just the first one?

**Comment by AlexAvlonitis** [comment]
yes that is the correct file.
This is what i gathered.
The first table is the usual results we see when we hit the details link (Row 1, 2...)
The other two tables are conditional, they will only appear if some conditions are met.
The second table will appear only if annotations are available and from what i saw this is something we are not using yet, i also checked the db and there were 0 annotations in there.
The third table i believe is for debugging purposes, not meant for the end users.
So i'm guessing you can only focus on the first table for now.
someone correct me if i am  wrong please.

**Comment by AlexAvlonitis** [comment]
@AlOneill updated: https://github.com/FreeUKGen/MyopicVicar/commit/22698d7e20e1f4c82a7151ce0bd5610d3f649b75

i deployed to test3. You can view the changes

**Comment by Captainkirkdawson** [comment]
Alex; your analysis is correct in terms of tables.

**Comment by Captainkirkdawson** [comment]
I do not like the field size dependant table display as it makes the header information cover too many lines

![display](https://cloud.githubusercontent.com/assets/2874694/21081083/43119790-bf7c-11e6-812d-e5910bb7eab4.JPG)

**Comment by AlOneill** [comment]
Kirk, the caption is my fault. The styling is the same as for the search results table. Table cells can be up to 32em (about 65 characters) wide -- for a long Notes field, say.

If we simply allow the table to expand to fill the width of the page (as previously), then the large horizontal spaces make for less comfortable scanning. I will ponder whether some sort of compromise is practicable.

If we take those two sentences out of the caption, there is then the dilemma of where and how to place them so that they align with something else. Bear in mind that centred text is hard to read for some people, especially when the lines are more than 65 or so characters wide.

Any concrete suggestions welcome!

**Comment by AlOneill** [comment]
Have pondered. On wider screens, the table cells can take a little more padding, which would typically reduce the number of caption lines by one (keeping the lines to a comfortable reading length).

If that is not enough, we could consider dropping the first sentence, "For information about this place ...", replacing it with a link to the place in the table itself.

![record_details](https://cloud.githubusercontent.com/assets/14852554/21107132/905f6600-c088-11e6-9a9e-eb13a66182cc.png)

**Comment by Captainkirkdawson** [comment]
Like some of your pondering. 
I like the idea of linking the place name in the table. We would have to have a note in the field
A couple of other ideas/questions.
Do we need the search date in the caption?
What about putting the reporting of an possible error as a footnote to the table?

**Comment by AlOneill** [comment]
(Apologies for delay, did not get notified of your comment.)

Not sure I understand "We would have to include a note in the field". Was thinking that as most of the fields are added to the table en bloc, we would need to extract each field_name in turn, then do "if field_name == Place, then make link" on the values. There is probably a better way ...

If my memory is correct, I think the search date was added in response to researcher requests.

Can't see how to set up a footnote so that text is *not* centred, yet it aligns with the table -- not with a centred table anyway.

**Comment by Captainkirkdawson** [comment]
Sorry was less than clear. What I was trying to say was that in the place field we need a note saying that clicking on the place name will provide more information about the place. ie the explanation that is currently in the caption

**Comment by AlOneill** [comment]
Ah, of course (doh). *If* explanation is needed, a title attribute on the link would not be accessible to keyboard-only users, so would need a tooltip -- extra text in Place field would probably annoy those who want to save the record for their future reference.

Let the scrum decide?

**Comment by AlOneill** [comment]
To clarify, a tooltip would also place extra text in the Place value field once printable format button is used.

**Comment by richpomfret** [comment]
Span the two columns so they are a single column.

**Comment by Captainkirkdawson** [comment]
Date to only be on printable version; link to name in list not chapeau; error report link to be in table

**Comment by AlOneill** [comment]
Error report link should not be in the table -- not semantic (so confusing for screen-reader user).

**Comment by Captainkirkdawson** [comment]
Attempted to respond to scrum discussion and subsequent concern. Available on test3 for review.

![detail display](https://cloud.githubusercontent.com/assets/2874694/21203061/01776000-c20e-11e6-9605-60faa152af16.JPG)

**Comment by Captainkirkdawson** [comment]
Printable version has search date
![printable version](https://cloud.githubusercontent.com/assets/2874694/21203172/8b52ae60-c20e-11e6-8a0a-aaa1e59d6f22.JPG)

**Comment by Sherlock21** [comment]
The on screen view is fine but could you mane the 2 green link fields a bot stronger colour please or slightly bigger font size or something. for those with advancing years, its hard to tell between the grey and free at this font size and shades. 

And yes, i just got my new glasses this week!

**Comment by Sherlock21** [comment]
And on the Search Results screen ( from which the page copied above) comes, Why was the navigation link text  changed?  What was wrong with "details"?  it was obvious: if you wanted details, you click on the word details.
 if the column needed a better title, then put "Link" don't trow out the baby with the batch water. the new navigation link invitation of " Row n " is meaningless and ridiculous

**Comment by AlOneill** [comment]
Suggest 'With' should be 'with' in 'Place (...)'

But, if 'Report Data error' needs no explanation, then does Place link need one?

(BTW, although subscribed to this thread, am still not getting notified about new comments. Sigh.)

**Comment by Captainkirkdawson** [comment]
Adjusted both texts; understand what you are saying about inconsistency but
do think they are different and given our audience !!!!
See new version

Kirk Dawson
5220 Riverside Drive
Fairmont Hot Springs, B.C.
V0B 1L1

**Comment by AlOneill** [comment]
Just noticed 'search conduction' on printable format. I would remove conduction: *if* a word is needed, suggest 'made'.

**Comment by Captainkirkdawson** [comment]
Good catch; corrected

Kirk Dawson
5220 Riverside Drive
Fairmont Hot Springs, B.C.
V0B 1L1
